src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -213,6 +213,54 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_lw_request_cc_val(void)
         }                                                   \
     } while (0)
 
+#ifndef HAVE_DEBUGGER_SUPPORT
+#define MPIDI_OFI_RECV_REQUEST_CREATE_COMM(req, comm, context_offset)                              \
+    do {                                                                                           \
+        int c = 0;                                                                                 \
+        (req) = MPIDIU_comm_get_recv_req((comm), (context_offset));                                \
+        MPIR_cc_incr(&(req)->cc, &c);                                                              \
+        MPL_DBG_MSG_FMT(MPIDI_CH4_DBG_GENERAL, VERBOSE, (MPL_DBG_FDEST, " lw_recv_req[%d] cc=%d",  \
+                                                         (context_offset), c));                    \
+        MPIR_Request_add_ref((req));                                                               \
+    } while (0)
+#else
+#define MPIDI_OFI_RECV_REQUEST_CREATE_COMM(comm, req, context_offset)                              \
+    do {                                                                                           \
+        int c = 0;                                                                                 \
+        (req) = MPIR_Request_create(MPIR_REQUEST_KIND__RECV);                                      \
+        MPIR_ERR_CHKANDSTMT((req) == NULL, mpi_errno, MPIX_ERR_NOREQ, goto fn_fail, "**nomemreq"); \
+        MPIR_cc_incr(&(req)->cc, &c);                                                              \
```

Comment:
On the lightweight send implementation, the completion counter is always zero. We only modify the reference count to ensure the request does not go back to the allocation pool. Does that not work in this case?

Reply:
I thought some more about this and think I understand why the approach was taken. Essentially each operation increments the counter on this request, and they all complete as a group. Still, there some cases to avoid from the looks of the test results.

Reply:
I thought some more about this and think I understand why the approach was taken. Essentially each operation increments the counter on this request, and they all complete as a group. Still, there are some cases to avoid from the looks of the test results.